### ISSUE TITLE
New semantic analyzer: overriding __init__ with Tuple base class

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6446]
This generates a false positive:

```py
from typing import Tuple

c: C
class C(Tuple[int, str]):
    def __init__(self) -> None: pass  # Cannot overwrite NamedTuple attribute "__init__"
```

This error gets generated when processing typeshed builtins import cycle, so this is high priority.
